drakrun/drakrun/vm.py

```diff
@@ -55,6 +59,54 @@ def generate_vm_conf(install_info: InstallInfo, vm_id: int):
     log.info("Generated VM configuration for vm-{vm_id}".format(vm_id=vm_id))
 
 
+def get_all_vm_conf() -> list:
+    regex = re.compile(r'vm-(\d+)\.cfg')
+    config_dir = os.path.join(ETC_DIR, 'configs')
+    vm_ids = []
+
+    for f in os.listdir(config_dir):
+        reg = regex.search(f)
+
+        if reg is not None:
+            vm_ids.append(int(reg.group(1)))
+
+    return vm_ids
+
+
+def delete_vm_conf(vm_id: int) -> bool:
+    config_dir = os.path.join(ETC_DIR, 'configs')
+    return safe_delete(os.path.join(config_dir, f"vm-{vm_id}.cfg"))
+
+
+def get_restore_percentage(vm_id: int) -> bool:
+    install_info = InstallInfo.try_load()
+    if install_info is None:
+        return
+    time.sleep(3)
+    cur_mem = 0
+    safe_zero_count = 0
+    with tqdm(total=install_info.memory) as pbar:
+        while cur_mem != install_info.memory:
+            try:
+                proc = subprocess.run(f"xl list vm-{vm_id} | tail -n 1 | tr -s ' ' | cut -f 3 -d ' '", shell=True, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+            except subprocess.CalledProcessError:
+                break
+
+            try:
+                cur_mem = int(proc.stdout.decode())
+                if safe_zero_count == 10:
+                    break
+                if cur_mem == pbar.n:
+                    safe_zero_count += 1
+                else:
+                    pbar.update(cur_mem - pbar.n)
+                    if cur_mem == install_info.memory:
+                        break
+            except ValueError:
+                pass
+            time.sleep(1)
```

Comment:
I know it may look cool but the code complexity and using `time.sleep` don't seem to be worth it.
Please remove it.

Reply:
My focus wasn't on coolness actually, It was to show the progress of the restoration and making it user-friendly so that it doesn't look like the program is stuck on long-running operation. 

It works on a different thread and exits as soon as the restoration is complete, but for that thread to not take too much CPU, I had put `time.sleep`.

Reply:
In most cases this function will be called by the drakrun daemon running in the background.
This will probably (I don't know how tqdm handles this) cause the logs to contain some garbage.

Reply:
> In most cases this function will be called by the drakrun daemon running in the background.
> This will probably (I don't know how tqdm handles this) cause the logs to contain some garbage.

Yeah, Now I can understand the real issue. Fine, I Will remove this